docs/desired-state-specification.md

```diff
@@ -0,0 +1,136 @@
+### Desired State Representation
+A technology agnostic way of representing the target software state of the device is required to enable remote management of device software and configuration. The device software state model enables flexible representation of complex software topologies of a device as well as the related configuration. It also enable users and application developers to develop and deploy composite applications seamlessly based on device context changes.
+
+### Desired State Data Model
+
+The following table describes all supported properties and sections of the Desired State specification:
+
+| Property | Type | Description |
+| - | - | - |
+| **General properties** | | | |
+| baselines | JSON array | List of Desired State baselines |
+| domains | JSON array | List of Desired State specifications for the domain agents |
+| **Baseline properties** | | |
+| title | string | Title of the baseline |
+| description | string | Description of the baseline |
+| preconditions | string | List of comma-separated precondition expressions |

```

Comment:
Preconditions before applying this baseline?

Reply:
I See, my question would also be how this preconditions are expressed and what kind of conditions can be checked, e.g., what do I write to check the battery status? But I guess this API is part of the next versions.

Reply:
It would be great to have an explanation of how to express such conditions. But we can add this in a later PR. The example below is already helpful to get the general style of how conditions are written.